local-interchain/interchain/handlers/actions.go

```diff
@@ -216,21 +218,49 @@ func (a *actions) relayerCheck(w http.ResponseWriter, r *http.Request) error {
 	return err
 }
 
-func KillAll(ctx context.Context, ic *interchaintest.Interchain, vals map[string][]*cosmos.ChainNode, relayer ibc.Relayer, eRep ibc.RelayerExecReporter) {
-	if relayer != nil {
-		if err := relayer.StopRelayer(ctx, eRep); err != nil {
+// KillAllICTContainers kills all containers that are prefixed with interchaintest specific container names.
+// This is required as you can not ctrl+c kill a local-ic instance before the ic.Build(), else containers will remain running.
+// (Since the vals map does not have context to these containers yet - due to still being built)
+func KillAllICTContainers(ctx context.Context) {
+	cli, err := client.NewClientWithOpts(client.FromEnv)
+	if err != nil {
+		panic(err)
+	}
+
+	containers, err := cli.ContainerList(ctx, dockerapitypes.ContainerListOptions{})
+	if err != nil {
+		panic(err)
+	}
+
+	for _, container := range containers {
+
+		if len(container.Names) == 0 {
+			continue
+		}
+
+		name := strings.ToLower(container.Names[0])
+		name = strings.TrimPrefix(name, "/")
+
+		// leave non ict relayed containers running
+		if !(strings.HasPrefix(name, "ict-") || strings.HasPrefix(name, "interchaintest-") || strings.HasPrefix(name, "ictrelay-")) {
```

Comment:
It might make sense to pull out "ict-" "interchaintest-" and "ictrelayer-" as constants somewhere. That should help limit typos or mistakes if we were ever to rename the containers. We could also use those constants when we create the containers.

Reply:
1897a72fec60a4f4e8e8f06694747a6287343289